docs/api_guide/plot_06_sklearn_pipeline_cv_demo.py

```diff
@@ -0,0 +1,473 @@
+#
+# To set up a scikit-learn `Pipeline`, ensure that:
+#
+# 1. Each intermediate step is a [scikit-learn transformer object](https://scikit-learn.org/stable/data_transforms.html) with a `transform` and/or `fit_transform` method.
+# 2. The final step is either another transformer or an [estimator object](https://scikit-learn.org/stable/developers/develop.html#estimators) with a `fit` method, or a model with `fit`, `predict`, and `score` methods.
+#
+# Each transformation step takes a 2D array `X` of shape `(num_samples, num_original_features)` as input and outputs another 2D array of shape `(num_samples, num_transformed_features)`. The final step takes a pair `(X, y)`, where `X` is as before, and `y` is a 1D array of shape `(n_samples,)` containing the observations to be modeled.
+#
+# You can define a pipeline as follows:
+# ```python
+# from sklearn.pipeline import Pipeline
+#
+# # Assume transformer_i/predictor is a transformer/model object
+# pipe = Pipeline(
+#     [
+#         ("label_1", transformer_1),
+#         ("label_2", transformer_2),
+#         ...,
+#         ("label_n", transformer_n),
+#         ("label_model", model)
+#     ]
+# )
+# ```
+#
+# Calling `pipe.fit(X, y)` will perform the following computations:
+# ```python
+# # Chain of transformations
+# X1 = transformer_1.fit_transform(X)
+# X2 = transformer_2.fit_transform(X1)
+# # ...
+# Xn = transformer_n.fit_transform(Xn_1)
+#
+# # Fit step
+# model.fit(Xn, y)
+# ```
+# And the same holds for `pipe.score` and `pipe.predict`.
+#
+# ## Why pipelines are useful
+#
+# Pipelines not only streamline and simplify your code but also offer several other advantages. The real power of pipelines becomes evident when combined with the scikit-learn `model_selection` module. This combination allows you to tune hyperparameters at each step of the pipeline in a straightforward manner.
+#
+# In the following sections, we will showcase this approach with a concrete example: selecting the appropriate basis type and number of bases for a GLM regression in NeMoS.
+#
+# ## Combining basis transformations and GLM in a pipeline
+# Let's start by creating some toy data.
+
+import nemos as nmo
+import numpy as np
+import pandas as pd
+import scipy.stats
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+from sklearn.pipeline import Pipeline
+from sklearn.model_selection import GridSearchCV
+
+# predictors, shape (n_samples, n_features)
+X = np.random.uniform(low=0, high=1, size=(1000, 1))
+# observed counts, shape (n_samples,)
+rate = 2 * (
+    scipy.stats.norm.pdf(X, scale=0.1, loc=0.25)
+    + scipy.stats.norm.pdf(X, scale=0.1, loc=0.75)
+)
+y = np.random.poisson(rate).astype(float).flatten()
+
+# %%
+fig, ax = plt.subplots()
+ax.scatter(X.flatten(), y, alpha=0.2)
+ax.set_xlabel("input")
+ax.set_ylabel("spike count")
+sns.despine(ax=ax)
+
```

Comment:
should describe this data: this is showing a simulated neuron's tuning curve, which here is bimodal, normally distributed around the peaks of 0.25 and 0.75.